### DISCUSSION TITLE
Config issue

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#23737] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

The renovated repo has two branches:  rhel-9.2 and rhel-9.3.
The parent image has tags: 9.2-x, 9.3-x'
How can I pick up the latest 9.2-x tag for the rhel-9.2 branch while the latest 9.3-x' is picked up for the rhel-9.3 branch?

### What you tried so far.

from config.js:
```
packageRules: [
    {
      matchDatasources: ["docker"],
      matchManagers: ["dockerfile"],
      registryUrls: ["quay.io/"],
      versioning: "redhat"
    }
]
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
you probably need two more rules with match base branches and allowed versions

**Reply by lmilbaum** [reply]
Thanks for the hint. Sounds right.

**Reply by lmilbaum** [reply]
I can't figure out how to limit the allowed versions. 
The regex is defined by the `versioning` key.
Should I use the `allowedVersions`? If yes, how?

**Reply by viceice** [reply]
something like:
```js
packageRules: [
    {
      matchDatasources: ["docker"],
      matchManagers: ["dockerfile"],
      registryUrls: ["quay.io/"],
      versioning: "redhat"
    },
   {
      matchDatasources: ["docker"],
      matchManagers: ["dockerfile"],
      matchBaseBranches: ["rhel-9.2"],
      allowedVersions: "/^9\\.2(?:-d+)?$/"
    },
   {
      matchDatasources: ["docker"],
      matchManagers: ["dockerfile"],
      matchBaseBranches: ["rhel-9.3"],
      allowedVersions: "/^9\\.3(?:-d+)?$/"
    }
]
```

**Reply by lmilbaum** [reply]
Nice. Thanks @viceice.
